File: plugins/setup/src/main/java/net/elytrarace/setup/conversation/map/MapAuthorPrompt.java
package net.elytrarace.setup.conversation.map;

import net.elytrarace.api.conversation.ConversationContext;
import net.elytrarace.api.conversation.Prompt;
import net.elytrarace.api.conversation.StringPrompt;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.JoinConfiguration;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.stream.Stream;

public class MapAuthorPrompt extends StringPrompt {
    @Override
    public @NotNull Component getPromptText(@NotNull ConversationContext context) {
        return Component.translatable("prompt.map.author");
    }

    @Override
    public @Nullable Prompt acceptInput(@NotNull ConversationContext context, @Nullable String input) {
        if (input == null) {
            return this;
        }
        if (input.isEmpty()) {
            return this;
        }
        var trimmedInput = input.trim();
        var authorList = Stream.of(trimmedInput.split(","))
                .map(String::trim)
                .map(MiniMessage.miniMessage()::deserialize).toList();
        context.setSessionData("author", Component.join(JoinConfiguration.commas(false), authorList));
        return new MapWorldPrompt();
    }
}

File: src/main/kotlin/net/elytrarace/conversation/MapAuthorPrompt.kt
package net.elytrarace.conversation

import net.elytrarace.model.dbo.Cup
import net.elytrarace.model.dbo.Cups
import net.elytrarace.model.dbo.ElytraMap
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.minimessage.MiniMessage
import org.jetbrains.exposed.sql.select
import org.jetbrains.exposed.sql.transactions.transaction

class MapAuthorPrompt : StringPrompt() {
    override fun getPromptText(context: ConversationContext): Component {
        return MiniMessage.miniMessage().deserialize("<lang:prompt.map.author>")
    }

    override fun acceptInput(context: ConversationContext, input: String?): Prompt? {
        context.setSessionData("map_author", input)
        transaction {
            val cup = Cups.select { Cups.name eq context.getSessionData("map_cup_name") as String }.firstOrNull() ?: return@transaction null
            ElytraMap.new {
                this.cup = Cup.wrapRow(cup)
                author = input
                world = context.getSessionData("map_world") as String
                name = context.getSessionData("map_name") as String
                displayName = context.getSessionData("map_name_display") as String
            }
        }
        return FinishPrompt()
    }
}